### ISSUE TITLE
Diagnostic suggestions highlighting can become misaligned

### ISSUE THREAD

**Issue by Alexendoo** [rust-lang/rust#104150]
Given a `cargo +nightly clippy` on the following:

```rust
#[derive(Copy, Clone)]
struct Foo(bool);

impl Into<(Foo, Foo)> for Foo {
    fn into(self) -> (Foo, Foo) {
        (self, self)
    }
}

fn main() {}
```

The current output is:

![image](https://user-images.githubusercontent.com/1830331/200573246-4d43a72d-aa69-46cf-b10c-f5f4e1bc0959.png)

In line 6, `(val, val)` the second highlight is misaligned, it's highlighted at `(val, v~al)~`

**Comment by NotStirred** [comment]
@rustbot release-assignment

Releasing as I am unable to test my changes to the compiler with clippy, perhaps due to #81431
If anyone can provide any guidance I'd be grateful

I believe the issue is in `rustc_errors::CodeSuggestion::splice_lines`, but can't verify without testing

**Comment by NotStirred** [comment]
This appears to be fixed on latest master
![image](https://user-images.githubusercontent.com/16853282/233853511-327f7a2b-fa54-4873-995e-c5ae0d0af6c5.png)
Though I am unsure of which commit fixed it

**Comment by Alexendoo** [comment]
Nice, looks like it was fixed by #108627
